test/text/write_test.cc

```diff
@@ -697,6 +697,21 @@ TEST(TextWriteTest, FunctionInlineImport) {
                        InlineExportList{InlineExport{Text{"\"m\""_sv, 1}}}});
 }
 
+TEST(TextWriteTest, Function_OmitFinalEnd) {
+  ExpectWrite(
+      "(func\n  nop\n  nop)"_sv,
+      Function{{}, {}, InstructionList{I{O::Nop}, I{O::Nop}, I{O::End}}, {}});
+}
+
+
+TEST(TextWriteTest, Function_DontOverDedent) {
+  // Multiple ends like this is syntatically malformed, but still should be
```

Comment:
Is this syntactically malformed?   Isn't more like semantically invalid?

Reply:
I think it's syntactically malformed, see the [control instructions](https://webassembly.github.io/spec/core/text/instructions.html#control-instructions) section, where for example a `block` instruction must be bracketed with the `end` keyword.